### PR TITLE
Fix mistakes with binary trees & spectral norm

### PR DESCRIPTION
**Pull request by OverloadedOrama**
The spectral norm algorithm benchmark implemented in #68 had the wrong inputs. My bad! 😅

EDIT: I also improved its static typing, making it run quite faster.

### CODE REVIEW

**Comment by OverloadedOrama** [comment]
I also had made a mistake with the binary trees algorithm porting and the results were different than the ones from other languages, which I fixed in this PR. That resulted in the algorithm being even slower now, so I lowered the inputs.

File: benchmarks/gdscript/binary_trees.gd [hunk 1]
```diff
@@ -7,14 +7,14 @@ extends Benchmark
 func bottom_up_tree(depth: int) -> Array:
 	if depth > 0:
 		depth -= 1
-		var left = bottom_up_tree(depth)[0]
-		var right = bottom_up_tree(depth)[1]
+		var left = bottom_up_tree(depth)
+		var right = bottom_up_tree(depth)
```
Comment (Calinou) [inline benchmarks/gdscript/binary_trees.gd:11]:
```suggestion
		var left := bottom_up_tree(depth)
		var right := bottom_up_tree(depth)
```

File: benchmarks/gdscript/spectral_norm.gd [hunk 2]
```diff
@@ -9,52 +9,55 @@ func eval_a(i: int, j: int) -> float:
...
-	var t := {}
+	var u := PackedFloat64Array([])
+	u.resize(n)
+	var v := PackedFloat64Array([])
+	v.resize(n)
+	var t := PackedFloat64Array([])
+	t.resize(n)
 	for i in n:
-		u[i] = 1
+		u[i] = 1.0
```
Comment (Calinou) [inline benchmarks/gdscript/spectral_norm.gd:41]: I think you can use `fill()` method here to immediately fill the array with the right value. I don't remember if it exists with packed arrays though.

Comment (OverloadedOrama) [reply benchmarks/gdscript/spectral_norm.gd]: It does indeed exist, I updated the code.

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**